### PR TITLE
Updated Documentation and FAQ

### PR DESCRIPTION
**Pull request by tweekerz**
Updated Documentation and FAQ

### CODE REVIEW

File: docs/Security/Extended-Protection.md [hunk 1]
```diff
@@ -282,10 +342,28 @@ If you want to enable Extended Protection in your environment manually without u
...
+
+You will also notice that some Health Mailbox logins fail with event ID: 4625 and failure reason "_An Error occurred during Logon_" and status _0xC000035B_ which is related to the failed probes. [**Get-ServerHealth**](https://docs.microsoft.com/en-us/exchange/high-availability/managed-availability/health-sets?view=exchserver-2019#use-the-exchange-management-shell-to-view-a-list-of-monitors-and-their-current-health) command will also show RPC and Mapi monitors as Unhealthy.
+
+**Impact of these failures**: Due to this probe failure, the Mapi and Rpc App pools will get restarted once. There should be no other impact.
+
+You can also turn off any of the above probes temporarily (till the fix is provided) by going through steps mentioned in [Configure managed availability overrides \| Microsoft Docs](https://docs.microsoft.com/en-us/exchange/high-availability/managed-availability/configure-overrides?view=exchserver-2019).
+
+Microsoft is working on a fix and will provide more information soon.
+
+1.  Issue with accessing Admin Audit logs
```
Comment (lusassl-msft) [inline docs/Security/Extended-Protection.md:357]: Not sure if this should already be posted. We should remove it until we have more information to publish.

File: docs/Security/Extended-Protection.md [hunk 2]
```diff
@@ -282,10 +342,28 @@ If you want to enable Extended Protection in your environment manually without u
...
+2. In Exchange Server 2013, 2016 and 2019 the following probes will show _FAILED_ status after running the script which switches on Extended Protection with required SSL flags on various vDirs as per recommended guidelines:
+   1. OutlookMapiHttpCtpProbe
+   2. OutlookRpcCtpProbe
+   3. OutlookRpcDeepTestProbe
+   4. OutlookRpcSelfTestProbe
+   5. ComplianceOutlookLogonToArchiveMapiHttpCtpProbe
+   6. ComplianceOutlookLogonToArchiveRpcCtpProbe
+
+You will also notice that some Health Mailbox logins fail with event ID: 4625 and failure reason "_An Error occurred during Logon_" and status _0xC000035B_ which is related to the failed probes. [**Get-ServerHealth**](https://docs.microsoft.com/en-us/exchange/high-availability/managed-availability/health-sets?view=exchserver-2019#use-the-exchange-management-shell-to-view-a-list-of-monitors-and-their-current-health) command will also show RPC and Mapi monitors as Unhealthy.
+
```
Comment (lusassl-msft) [inline docs/Security/Extended-Protection.md:350]: I’ve added the PS commands to the doc (in review mode). We should add them here. To be confirmed by @mukeshHCW

**Review by lusassl-msft — DISMISSED** [review]

File: docs/Security/Extended-Protection.md [hunk 3]
```diff
@@ -199,12 +199,16 @@ It’s not required to run the script directly on any specific Exchange Server i
...
+| SkipExchangeServerNames       | Used to specify which Exchange servers should be **excluded from the scope of script execution**. It can be either a single Exchange Server hostname or a comma separated list of hostnames. Parameter values: **Exchange Server Hostname (NetBIOS or FQDN)**                                                                                                                                |
+| RollbackType                  | Used to revert changes made by the Extended Protection script. Parameter Values: **"RestoreIISAppConfig**"                                                                                                                                                                                                                                                                                   |
```
Comment (dpaulson45) [inline docs/Security/Extended-Protection.md:200]: Also Rollback type of `RestrictTypeEWSBackend`

**Review by bill-long — DISMISSED** [review]

File: docs/Security/Extended-Protection.md [hunk 4]
```diff
@@ -222,6 +226,48 @@ After the prerequisites checks have been passed, the script will enable Extended
 
 ![Text Description automatically generated](attachments/9a4e6863e860e064d2831d7d714c95ce.png)
 
+#### **Scenario 2**: Using Retention Policies for automated archiving of mailboxes
+
+If you are using a _Retention Policy_ containing _Retention Tags_ which perform _Move to Archive_ actions, you should use the script as follows:
+
+.\\ExchangeExtendedProtectionManagement.ps1 -RestrictType EWSBackend -IPRangeFilePath "IPList.txt"
```
Comment (dpaulson45) [inline docs/Security/Extended-Protection.md:233]: Not in a code block

File: docs/Security/Extended-Protection.md [hunk 5]
```diff
@@ -222,6 +226,48 @@ After the prerequisites checks have been passed, the script will enable Extended
...
+
+This list of addresses should be provided in a TXT file in this way:
+
+10.68.12.48
+
+2001:4898:5808:14:3938:7838:67ab:cdfb
+
+10.128.4.204
+
+10.231.234.140/20
```
Comment (dpaulson45) [inline docs/Security/Extended-Protection.md:249]: Place in code block

File: docs/Security/Extended-Protection.md [hunk 6]
```diff
@@ -222,6 +226,48 @@ After the prerequisites checks have been passed, the script will enable Extended
...
+
+File should contain only IP addresses or IP subnets of Exchange servers separated by newline character. For customers who have more than 100 servers, it is recommended to use IP subnets instead of IP addresses, so that the list is short and efficient.
+
+To help get all the IP addresses of all Exchange Servers in the organization, customers can use the following command.
+
+.\\ExchangeExtendedProtectionManagement.ps1 -FindExchangeServerIPAddresses
```
Comment (dpaulson45) [inline docs/Security/Extended-Protection.md:255]: code block

File: docs/Security/Extended-Protection.md [hunk 7]
```diff
@@ -222,6 +226,48 @@ After the prerequisites checks have been passed, the script will enable Extended
...
+
+The above command will generate a file IPList.txt in the directory where script is stored.
+
+**Important Note:** Customers with many Exchange Servers are expected to modify this list and use IP subnets to reduce the number of allow rules to be added.
+
+#### Scenario 3: Using Modern Hybrid Configuration or Hybrid Agent
+
+In case you have Modern Hybrid configuration, you need to skip Exchange servers published using the Hybrid Agent. This can be done by using the _SkipExchangeServerNames_ parameter:
+
+.\\ExchangeExtendedProtectionManagement.ps1 -SkipExchangeServerNames HybridServer1, HybridServer2
```
Comment (dpaulson45) [inline docs/Security/Extended-Protection.md:265]: code block

File: docs/Security/Extended-Protection.md [hunk 8]
```diff
@@ -222,6 +226,48 @@ After the prerequisites checks have been passed, the script will enable Extended
...
+
+Or
+
+.\\ExchangeExtendedProtectionManagement.ps1 -RestrictType EWSBackend -IPRangeFilePath "IPList.txt" -SkipExchangeServerNames HybridServer1, HybridServer2
```
Comment (dpaulson45) [inline docs/Security/Extended-Protection.md:269]: code block

File: docs/Security/Extended-Protection.md [hunk 9]
```diff
@@ -244,7 +290,21 @@ After the prerequisites checks have been passed, the script will enable Extended
...
+You can also use the script to roll back the **Extended Protection settings and any IP restriction rules** added via script from one or more servers. When Extended Protection settings are modified by the script, an applicationHost.cep.\*.bak file is created on each server, which contains a backup of pre-existing settings before the script is run. Those files are going to be local to each individual server that the script modifies. Therefore, the rollback of Extended Protection settings can be rolled back from any machine where the script will run using the _earliest_ version of the .bak file to roll back the changes.
+
+The following command initiates a full rollback of **Extended Protection settings** and **IP restriction rules** on any Exchange server where it was enabled using the script:
+
+.\\ExchangeExtendedProtectionManagement.ps1 –RollbackType RestoreIISAppConfig
```
Comment (dpaulson45) [inline docs/Security/Extended-Protection.md:297]: code block

File: docs/Security/Extended-Protection.md [hunk 10]
```diff
@@ -244,7 +290,21 @@ After the prerequisites checks have been passed, the script will enable Extended
...
+
+#### Roll back of IP Restrictions
+
+You can use the script to **only** roll back **Allow and Deny rules** set in Backend EWS vdir’s IP Address and Domain Restriction module in the following way.
+
+.\\ExchangeExtendedProtectionManagement.ps1 -RollbackType RestrictTypeEWSBackend
```
Comment (dpaulson45) [inline docs/Security/Extended-Protection.md:303]: code block

File: docs/Security/Extended-Protection.md [hunk 11]
```diff
@@ -244,7 +290,21 @@ After the prerequisites checks have been passed, the script will enable Extended
...
+
+**Note:** To safeguard Backend EWS vdir against NTLM relay, executing above command will set Extended Protection setting back to Required. If automated archiving is enabled, executing this command can cause automated archiving to stop working till the time Allow and Deny rules are not added back using the following command.
+
+.\\ExchangeExtendedProtectionManagement.ps1 -RestrictType EWSBackend -IPRangeFilePath "IPList.txt"
```
Comment (dpaulson45) [inline docs/Security/Extended-Protection.md:307]: code block

**Review by dpaulson45 — CHANGES_REQUESTED** [review]

**Review by dpaulson45 — APPROVED** [review]

**Review by lusassl-msft — APPROVED** [review]

**Pull request merged**